ucte/ucte-converter/src/main/java/com/powsybl/ucte/converter/UcteExporter.java

```
@@ -0,0 +1,1096 @@
+    }
+
+    /**
+     * @param twoWindingsTransformer The twoWindingsTransformer containing the PhaseTapChanger we want to convert
+     * @return P (MW) of the angle regulation for the two windings transformer
+     */
+    private float calculateAngleP(TwoWindingsTransformer twoWindingsTransformer) {
+        return (float) twoWindingsTransformer.getPhaseTapChanger().getRegulationValue();
+    }
+
+    /**
+     * Give the type of the UcteAngleRegulation
+     *
+     * @param twoWindingsTransformer containing the PhaseTapChanger we want to convert
+     * @return The type of the UcteAngleRegulation
+     */
+    private UcteAngleRegulationType findRegulationType(TwoWindingsTransformer twoWindingsTransformer) {
+        if (isSymm(twoWindingsTransformer)) {
+            return UcteAngleRegulationType.SYMM;
+        } else {
+            return UcteAngleRegulationType.ASYM;
+        }
+    }
+
+    private boolean isSymm(TwoWindingsTransformer twoWindingsTransformer) {
+        for (int i = twoWindingsTransformer.getPhaseTapChanger().getLowTapPosition();
+             i < twoWindingsTransformer.getPhaseTapChanger().getHighTapPosition(); i++) {
+            if (twoWindingsTransformer.getPhaseTapChanger().getStep(i).getRho() != 1) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    public UcteNodeCode createUcteNodeCode(String id, VoltageLevel voltageLevel, String country) {
+        UcteNodeCode ucteNodeCode;
+        if (iidmIdToUcteNodeCodeId.containsKey(id)) {
+            return iidmIdToUcteNodeCodeId.get(id);
+        }
+        Optional<UcteNodeCode> optionalUcteNodeCode = parseUcteNodeCode(id);
+        if (optionalUcteNodeCode.isPresent()) { // the ID is already an UCTE id
+            iidmIdToUcteNodeCodeId.put(id, optionalUcteNodeCode.get());
+            return optionalUcteNodeCode.get();
+        } else {
+            convertIidmIdToUcteNodeCode(id, voltageLevel, country);
```

Should we throw an exception instead?